gatsby-node.js

```diff
@@ -40,3 +44,110 @@ exports.onCreateWebpackConfig = ({ stage, actions, getConfig }) => {
     actions.replaceWebpackConfig(config)
   }
 }
+
+// const checkIfDirExists = async folderPath => {
+//   try {
+//     await stat(folderPath)
+//     return true
+//   } catch (err) {
+//     if (err.code === 'ENOENT') {
+//       return false
+//     }
+//   }
+// }
+
+// const mkDirIfItDoesNotExist = async folderPath => {
+//   const doesDirExist = await checkIfDirExists(folderPath)
+//   if (!doesDirExist) {
+//     return mkdir(folderPath)
+//   }
+// }
+
+// exports.onPostBuild = async ({ graphql }) => {
+//   try {
+//     const {
+//       data: {
+//         allBlogPost: { nodes }
+//       }
+//     } = await graphql(`
+//       query GetBlogMainImages {
+//         allBlogPost {
+//           nodes {
+//             picture {
+//               gatsbyImageData(width: 850)
+//             }
+//             date
+//           }
+//         }
+//       }
+//     `)
+
+//     await mkDirIfItDoesNotExist('./public/blog/images')
+
+//     await Promise.all(
+//       nodes.map(async node => {
+//         const date = node.date.slice(0, 10)
+
+//         return mkDirIfItDoesNotExist(`./public/blog/images/${date}`)
+//       })
+//     )
+
+//     await Promise.all(
+//       nodes.map(async node => {
+//         const imagePath = node.picture.gatsbyImageData.images.fallback.src
+//         const date = node.date.slice(0, 10)
+//         return copyFile(
+//           path.join(__dirname, 'public', imagePath),
+//           path.join(
+//             __dirname,
+//             '/public/blog/images',
+//             date,
+//             path.basename(imagePath)
+//           )
+//         )
+//       })
+//     )
+//   } catch (err) {
+//     console.error(err)
+//   }
+// }
+
+exports.onPostBuild = async ({ graphql }) => {
+  try {
+    const {
+      data: {
+        allBlogPost: { nodes }
+      }
+    } = await graphql(`
+      query GetBlogMainImages {
+        allBlogPost {
+          nodes {
+            picture {
+              fields {
+                sourcePath
+              }
+            }
+            date
+          }
+        }
+      }
+    `)
+
+    await Promise.all(
+      nodes.map(async node => {
+        const imagePath = node.picture.fields.sourcePath
+        return Jimp.read(path.join(__dirname, 'static', 'uploads', imagePath))
```

Comment:
Since I couldn't use `copyFile`, I used `Jimp` to take the main images and copy them over to the public folder. But if anyone has any ideas on getting the commented solution to work, it would be probably be better to just use `gatsby-plugin-image` and not using `Jimp` at all.